### PR TITLE
Fix outdated multi-period usage docs

### PR DESCRIPTION
**Pull request by jokochems**
* Correct the type of `periods` attribute of an energy system to list.
* Add hint to usage of `timeincrement` in combination with `timeindex`.
* Provide minor doc fixes alongside.

### CODE REVIEW

**Comment by jokochems** [comment]
@henhuy This should address the outdated docs. Thanks for pointing this out. Would you please review?

I really thought, I changed that in the first place after @p-snft pointed me to the possibility for code simplification. Seems like I either forgot or falsely commited to a wrong local branch that was never pushed back. Nevermind, now it should be correct.

File: src/oemof/solph/_energy_system.py [hunk 1]
```diff
@@ -176,12 +176,12 @@ def __init__(
...
     def _extract_periods_years(self):
-        """Map simulation years to the respective period based on time indices
+        """Map years in optimization to respective period based on time indices
 
         Returns
         -------
         periods_years: dict
-            the simulation year of the start of each a period,
+            year of the start of each period,
             relative to the start of the optimization run and starting with 0
```
Comment (henhuy) [inline src/oemof/solph/_energy_system.py:185]: Two things:
- function does not return anything, but sets member variable instead - so `Returns` section is not needed, right?
- if you want to use `Returns` section anyway or want to mention `period_years`, `periods_years` should be of type `list`

**Review by henhuy — CHANGES_REQUESTED** [review]
Minor issue found.
Thanks for adapting docs!

File: src/oemof/solph/_energy_system.py (hunk 1 above)
Comment (jokochems) [reply src/oemof/solph/_energy_system.py]: Thank you for pointing this out. I resolved it. Please have a look.

**Review by henhuy — APPROVED** [review]
Thanks for fixing the docs!

**Comment by jokochems** [comment]
@p-snft since this only affects docs, all tests are succeeding and @henhuy has already reviewed, I'll merge this.

**Pull request merged**